src/ripple/app/tx/impl/SetAccount.cpp

```diff
@@ -562,6 +589,14 @@ SetAccount::doApply()
             uFlagsOut &= ~lsfDisallowIncomingTrustline;
     }
 
+    // Set flag for clawback
+    if (ctx_.view().rules().enabled(featureClawback) &&
+        uSetFlag == asfAllowClawback)
+    {
+        JLOG(j_.trace()) << "set allow clawback";
+        uFlagsOut |= lsfAllowClawback;
+    }
```

Comment:
this flag is never meant to be cleared. Once set, it cannot be cleared

Reply:
this flag is never meant to be cleared. Once set, it cannot be cleared

Reply:
But why? This is the inverse of "no freeze" and it should definitely be a flag you can clear.

Reply:
@nbougalis The reason for making clawback opt-out by default was purely because, having clawback opt-in was a poor idea and would raise a lot of concerns, just like how having freeze opt-in by default was a bad choice. The verdict is that we would like "opt-out" these kind of regulatory features in the future, and clawback would be the first one to follow that. 

What is really the incentive for making allow clawback a mutable field?  If they really want to mutate it, then I don't see why can't they create a new account.

But I think it would be a good practice to stick with immutable fields for regulatory features like freeze or clawback.

Reply:
No, you misunderstand: I'm not suggesting that clawback should be opt-out, that would be a bad idea. I was just saying that I would allow it to be unset, but I don't feel strongly enough about it.

Reply:
I'm not quite sure if there is any need for this. But if we find out that unsetting the flag is really wanted somehow, I think it would it would be fine to add it through an amendment later. This would be a rather small change. But I think it would be a good idea to keep it simple for now.

Reply:
Going on a tangent here: Since `Clawback` and `NoFreeze` are complimentary to one another, is it a good idea to frame this condition as an invariant? I would eliminate such a check in other parts of the code right?

Reply:
I don't think this change would eliminate such checks in other parts of the code. Invariant check is executed _after_ a transaction.

And it will still be _mandatory_ to check this complementary property during the `preclaim` (before an transaction) of an `AccountSet`.